### PR TITLE
Edit duedate tff154

### PR DESCRIPTION
**Pull request by gearedmonkey**
### Summary

You can now change the due date of an existing checkout

### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. Start server on edit-duedate-tff154
2. Lookup student 123456
3. Checkout an item
4. Complete item checkout
5. Click change duedate button
6. Verify date can only be in future
7. Choose date.
8. Verify date has changed in the panel.

### Links

- [TFF-154](https://msoese.atlassian.net/browse/TFF-154)
- TheFourFifths/consus/pull/99

### CODE REVIEW

File: src/views/components/student-panel.jsx [hunk 1]
```diff
@@ -58,60 +62,106 @@ export default class StudentPanel extends ListenerComponent {
...
+
+    changeItemDueDate(date, item) {
+        StudentPanelController.changeItemDueDate(date, item);
+        this.setState({
+            showItemDateModal: false
+        })
+    }
+
+    changeModelDueDate(date, model) {
+        console.log('Model due dates not yet implemented.' + 'Changing due date to: ' + date + ' for model: ' + model.address);
```
Comment (mattmahn) [inline src/views/components/student-panel.jsx]: template string

File: src/controllers/components/student-panel.js [hunk 2]
```diff
@@ -1,14 +1,37 @@
...
+                Dispatcher.handleAction("STUDENT_FOUND", student);
+            }).catch(e => {
+                Dispatcher.handleAction('ERROR', { error: e.message });
+            });
+        }
+    }
 
+    static isValidDueDate(dueDate){
+        let today = moment();
+        let dueDateMoment = moment.tz(dueDate, 'America/Chicago');
```
Comment (mattmahn) [inline src/controllers/components/student-panel.js]: NACK: use config file for timezone

File: src/controllers/components/student-panel.js [hunk 3]
```diff
@@ -1,14 +1,37 @@
...
+        if(!dueDateMoment.isAfter(today)){
+            Dispatcher.handleAction('ERROR', {
+                error: 'Due date cannot be set to today or past.'
```
Comment (mattmahn) [inline src/controllers/components/student-panel.js]: Nit: "Due date must be in the future."

File: src/views/components/student-panel.jsx [hunk 4]
```diff
@@ -58,60 +62,106 @@ export default class StudentPanel extends ListenerComponent {
...
-            return (<div>
-                {model.name} {item.timestamp < Math.floor(Date.now()/1000) ? '(overdue)' : ''} <i>{item.address} Due on: {dueDate.format('MMMM Do YYYY, h:mm:ss a')}</i>
-            </div>)
+            return (
+                <div className="item-info">
+                    <div onClick={this.displayItem.bind(this, item.address)}
+                         className={item.timestamp < Math.floor(Date.now() / 1000) ? 'link-nostyle overdue' : 'link-nostyle'}>
+                        {model.name} {item.timestamp < Math.floor(Date.now() / 1000) ? '(overdue)' : ''}
+                        <i>{item.address}
+                            Due on: {dueDate.format('MMMM Do YYYY, h:mm:ss a')}</i>
```
Comment (tondef001) [inline src/views/components/student-panel.jsx]: Missing a space after the address

File: src/views/components/date-modal.jsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    render() {
+        return (<Modal
+                active={this.props.active}
+                buttonText="Confirm"
+                onClose={() => this.props.onDateSelected(this.state.date)}><p>{this.props.message}</p><br/><input
+                type="date" onChange={this.changeDate.bind(this)}/><br/>
+                <button onClick={() => this.props.onDateSelected(undefined)}>Cancel</button>
+            </Modal>
+        )
```
Comment (mattmahn) [inline src/views/components/date-modal.jsx]: `;`

File: src/controllers/components/student-panel.js (hunk 2 above)
Comment (gearedmonkey) [inline src/controllers/components/student-panel.js]: yesas

File: src/controllers/components/student-panel.js (hunk 3 above)
Comment (gearedmonkey) [inline src/controllers/components/student-panel.js]: Burn.

File: src/views/components/date-modal.jsx (hunk 5 above)
Comment (gearedmonkey) [inline src/views/components/date-modal.jsx]: Where do ;'s go to from line 27 after 2 days from the morrow of yesterdays moon cycle

File: src/views/components/student-panel.jsx (hunk 4 above)
Comment (gearedmonkey) [inline src/views/components/student-panel.jsx]: Where though

Comment (gearedmonkey) [inline src/views/components/student-panel.jsx]: I don't like this.

File: src/views/components/student-panel.jsx (hunk 1 above)
Comment (gearedmonkey) [inline src/views/components/student-panel.jsx]: No

File: src/views/components/date-modal.jsx (hunk 5 above)
Comment (Cafferoq) [inline src/views/components/date-modal.jsx]: What are you on about?

**Pull request merged**